presto-main/src/main/java/io/prestosql/execution/scheduler/SimpleNodeSelector.java

```diff
@@ -149,7 +153,79 @@ public SplitPlacementResult computeAssignments(Set<Split> splits, List<RemoteTas
             remainingSplits = splits;
         }
 
-        for (Split split : remainingSplits) {
+        Set<Split> remainingSplitsAfterBatchScheduling;
+        if (optimizedLocalScheduling && optimizedBatchScheduling) {
+            // only schedule split can be accessed remotely
+            remainingSplitsAfterBatchScheduling = remainingSplits.stream().filter(split -> !split.isRemotelyAccessible()).collect(Collectors.toSet());
```

Comment:
Normally we place each step of the stream processing on a separate like as follows:
```java
remainingSplits.stream()
        .filter(split -> !split.isRemotelyAccessible())
        .collect(Collectors.toSet());
``
We also normally use immutable collections unless we expect modification. In this case `toImmutableSet`